lib/migration.js

```diff
@@ -893,6 +894,31 @@ function mixinMigration(MySQL, mysql) {
     return columnType;
   }
 
+  function dateOptionsByType(p, columnType) {
+    switch (columnType.toLowerCase()) {
+      default:
+      case 'date':
+        break;
+      case 'time':
+      case 'datetime':
+      case 'timestamp':
```

Comment:
I think I don't understand your comment. As the MySQL documentation said: [TIME](https://dev.mysql.com/doc/refman/5.7/en/datetime.html), DATETIME and TIMESTAMP types can have a precision. The DATE-type (of course) not.

This function checks the columntype and for columns with a type of time, datetime and timestamp and a precision is defined, the columnType will modified (see line 915), for example to DATETIME(3). This will allow to save values like 2017-05-07 12:34:56.789. When the column is of the type DATE, the function returns the colomnType without any modification.